modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/components/PopupDropdownMenuTooltip.kt

```diff
@@ -0,0 +1,304 @@
+package au.com.shiftyjelly.pocketcasts.compose.components
+
+import android.content.res.Configuration
+import androidx.compose.animation.core.MutableTransitionState
+import androidx.compose.animation.core.animateFloat
+import androidx.compose.animation.core.rememberTransition
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Card
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Outline
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntRect
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Popup
+import androidx.compose.ui.window.PopupPositionProvider
+import androidx.compose.ui.window.PopupProperties
+import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
+import au.com.shiftyjelly.pocketcasts.compose.Devices
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import com.airbnb.android.showkase.annotation.ShowkaseComposable
+
+/**
+ * The Material3 TooltipBox comes with a couple of constraints:
+ * Both TooltipBox and RichTooltipBox lack an onDismissRequest property: https://issuetracker.google.com/issues/349864868
+ * It dismisses on tapping outside the component. Even if you re-display it, interaction with the rest of the screen is blocked until the tooltip is dismissed.
+ *
+ * This is a custom Tooltip implementation using Popup based on
+ * https://stackoverflow.com/a/69664787/193545
+ * It permits screen interaction even when the tooltip is visible and can be used with rest of the M2 components.
+ */
+
+@Composable
+fun PopupDropdownMenuTooltip(
```

Comment:
I just renamed this old tooltip component with a slightly different behavior used for referrals from `Tooltip` to `PopupDropdownMenuTooltip`.  Not sure why it is being shown as a new file. You can ignore reviewing it.